code/__HELPERS/game.dm

```diff
@@ -279,7 +279,7 @@ var/list/DummyCache = list()
 
 	//Next, check objects to block entry that are on the border
 	for(var/obj/border_obstacle in targetturf)
-		if((border_obstacle.flags & ON_BORDER) && (target != border_obstacle))
+		if(!(target.flags & ON_BORDER) && (border_obstacle.flags & ON_BORDER) && (target != border_obstacle))
```

Comment:
Yeah no that is not how this works, why didn't you just ask us? This should be done by uncross above and not cross.